dataline-integrations/singer/local_csv/destination/entrypoint.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e
+
+PROCESSED_CONFIG_FILE="processed_config.json"
+
+function echo2() {
+  echo >&2 "$@"
+}
+
+function error() {
+  echo2 "$@"
+  exit 1
+}
+
+function main() {
+  echo "hi!"
+  ARGS=
+  while [ $# -ne 0 ]; do
+    case "$1" in
+    -a | --discover)
+      ARGS="$ARGS --discover"
+      shift 1
+      ;;
+    -b | --config)
+      jq '.destination_path = "/local/" + .destination_path' $2 > $PROCESSED_CONFIG_FILE
+      ARGS="$ARGS --config $PROCESSED_CONFIG_FILE"
+      shift 2
+      ;;
+    *)
+      error "Unknown option: $1"
+      shift
+      ;;
+    esac
+  done
+
+  echo "ARGS"
+  echo $ARGS
+  if [[ "$ARGS" =~ .*"--discover".* ]]; then
+    echo2 "Discovering..."
+    DESTINATION_PATH=$(jq -r '.destination_path' $PROCESSED_CONFIG_FILE)
+    if [[ ! -d "$DESTINATION_PATH" && ! -L "$DESTINATION_PATH" ]] ; then
+      echo2 "Could not access path $DESTINATION_PATH" && exit 1
+    fi
+    # If connection check is successful write a fake catalog for the discovery worker to find
+    echo '{"streams":[]}' > "catalog.json"
```

Comment:
shouldn't that be written to stdout?

Reply:
huh. yeah. this isn't great. we have a few integrations doing this. i'll fix this one and then create an issue to fix other existing integrations this this problem.